plugins/com.google.cloud.tools.eclipse.sdk.test/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManagerTest.java

```diff
@@ -39,4 +52,149 @@ public void testManagedSdkOption_featureForced() {
+  @Test
+  public void testPreventModifyingSdk_cannotWrite() throws InterruptedException {
+    CloudSdkManager.preventModifyingSdk();
+    try {
+      assertFalse(CloudSdkManager.modifyLock.writeLock().tryLock());
+    } finally {
+      CloudSdkManager.allowModifyingSdk();
+    }
+  }
+
+  @Test
+  public void testPreventModifyingSdk_canRead() throws InterruptedException {
+    CloudSdkManager.preventModifyingSdk();
+    try {
+      Lock readLock = CloudSdkManager.modifyLock.readLock();
+      assertTrue(readLock.tryLock());
+      readLock.unlock();
+    } finally {
+      CloudSdkManager.allowModifyingSdk();
+    }
+  }
+
+  @Test
+  public void testAllowModifyingSdk_allowsWrite() throws InterruptedException {
+    CloudSdkManager.preventModifyingSdk();
+    CloudSdkManager.allowModifyingSdk();
+
+    Lock writeLock = CloudSdkManager.modifyLock.writeLock();
+    assertTrue(writeLock.tryLock());
+    writeLock.unlock();
+  }
+
+  @Test
+  public void testPreventModifyingSdk_doesNotBlockSimultaneousCalls() throws InterruptedException {
+    CloudSdkManager.preventModifyingSdk();
+
+    try {
+      Job job = new Job("another caller") {
+        @Override
+        public IStatus run(IProgressMonitor monitor) {
+          try {
+            CloudSdkManager.preventModifyingSdk();
+            return Status.OK_STATUS;
+          } catch (InterruptedException e) {
+            return Status.CANCEL_STATUS;
+          } finally {
+            CloudSdkManager.allowModifyingSdk();
+          }
+        }
+      };
+      job.schedule();
+      job.join();
+
+      assertTrue(job.getResult().isOK());
+    } finally {
+      CloudSdkManager.allowModifyingSdk();
+    }
+  }
+
+  @Test
+  public void testPreventModifyingSdk_blocksRunInstallJob() throws InterruptedException {
+    CloudSdkManager.preventModifyingSdk();
+    boolean prevented = true;
+
+    try {
+      final BaseCloudSdkInstallJob installJob = new FakeInstallJob(Status.OK_STATUS);
+
+      Job concurrentLauncher = new Job("concurrent thread attempting runInstallJob()") {
+        @Override
+        public IStatus run(IProgressMonitor monitor) {
+          try {
+            CloudSdkManager.runInstallJob(null, installJob);
+            return Status.OK_STATUS;
+          } catch (CoreException | InterruptedException e) {
+            return Status.CANCEL_STATUS;
+          }
+        }
+      };
+      concurrentLauncher.schedule();
+
+      while (installJob.getState() != Job.RUNNING) {
+        Thread.sleep(50);
+      }
+      // Incomplete test, but if it ever fails, something is surely broken.
+      assertEquals(Job.RUNNING, concurrentLauncher.getState());
+
+      CloudSdkManager.allowModifyingSdk();
+      prevented = false;
+      concurrentLauncher.join();
+
+      // Incomplete test, but if it ever fails, something is surely broken.
+      assertTrue(installJob.getResult().isOK());
+      assertTrue(concurrentLauncher.getResult().isOK());
+    } finally {
+      if (prevented) {
```

Comment:
The logic here is a little unclear to me. Why might prevented be true? I guess if an exception is thrown before we set prevented = false in which case the test fails?

Does it make sense to simply set allowModifyingSdk to true in an `@After` method instead?

Reply:
What you guessed is right. However, `allowModifyingSdk()` must match 1-to-1 with the number of `prevent...()`. That is why the Boolean var.